apps/wakunode2/config.nim

```diff
@@ -543,16 +539,19 @@ proc readValue*(r: var EnvvarReader, value: var crypto.PrivateKey) {.raises: [Se
 
 {.push warning[ProveInit]: off.}
 
-proc load*(T: type WakuNodeConf, version=""): ConfResult[T] =
+proc load*(T: type WakuNodeConf, version="", rng: ref HmacDrbgContext): ConfResult[T] =
   try:
-    let conf = WakuNodeConf.load(
+    var conf = WakuNodeConf.load(
       version=version,
       secondarySources = proc (conf: WakuNodeConf, sources: auto) =
         sources.addConfigFile(Envvar, InputFile("wakunode2"))
 
         if conf.configFile.isSome():
           sources.addConfigFile(Toml, conf.configFile.get())
     )
+    if $conf.nodekey == "":
+      conf.nodekey = crypto.PrivateKey.random(Secp256k1, rng[]).tryGet()
```

Comment:
This default value assignment should be in the wakunode2 module, like the CLI argument validations.

Reply:
Do you mean, after reading the config, we check if nodekey is `none`, and then assign a random one? 

Something like this? 

```nim
let nodekey = if conf.nodekey.isSome(): conf.nodekey.get()
              else: crypto.PrivateKey.random(Secp256k1, rng[]).tryGet()
```

Reply:
Addressed in 1b9566c8. Please let me know if this is what you thought of.

Reply:
Thanks!